### ISSUE TITLE
code widget contains many blank lines

### ISSUE THREAD

**Issue by yannickmoy** [AdaCore/learn#405]
The size of the widget used to display compilable code does not always fit the code it contains, sometimes it has a huge number of blank lines, see e.g. the third widget on the page:

https://learn.adacore.com/labs/intro-to-ada/chapters/imperative_language.html

**Comment by Robert-Tice** [comment]
This is a side effect of PR #333 (https://github.com/AdaCore/learn/pull/333).

The main.adb tab contains more code than the other tabs, which sets the minimum tab length to be the size of the main.adb tab code length. This is intended behavior.

**Comment by yannickmoy** [comment]
ok, looks weird but we can live with it!

**Comment by Robert-Tice** [comment]
I agree, it does look weird. We will keep this issue open as an enhancement to fix the UI problem created by the UX fix on #333
